File: src/main/java/com/pebble/baseAuth/domain/UserRepository.java
package com.pebble.baseAuth.domain;

import java.util.Optional;

public interface UserRepository {

    boolean existsByUsernameAndDeletedAtIsNull(String username);

    Optional<User> findByUsernameAndDeletedAtIsNull(String username);

    Optional<User> findByProviderAndProviderIdAndDeletedAtIsNull(String provider, String providerId);

    User save(User user);

}

File: src/main/kotlin/com/pebble/baseAuth/domain/UserRepository.kt
package com.pebble.baseAuth.domain

import java.util.Optional

interface UserRepository {

    fun existsByUsernameAndDeletedAtIsNull(username: String): Boolean

    fun findByUsernameAndDeletedAtIsNull(username: String): Optional<User>

    fun findByProviderAndProviderIdAndDeletedAtIsNull(provider: String, providerId: String): Optional<User>

    fun save(user: User): User

}
